.github/workflows/gh-ci.yml

```diff
@@ -231,7 +231,7 @@ jobs:
           rsync -R --files-from=artifacts.list . ${{ github.job }}-artifacts
           tar -zcvf ${{ github.job }}-artifacts.tar.gz ${{ github.job }}-artifacts
       - name: Upload Build Artifacts
-        if: success()
+        if: success() || failure()
```

Comment:
Is there a third state?  Why even have the if we'll do it on success or failure?

Reply:
> Is there a third state? 

There is `always()` but it gets executed even if we cancel the job.

> Why even have the if we'll do it on success or failure?

If used with no condition or with `success()`, it collects logs only when a build is successful. 

With `failure()`  logs are collected only on build failure, not when the build is successful.

We want to collect logs on either build failure or success but not when a job is manually canceled.